app/assets/javascripts/govuk_publishing_components/components/checkboxes.js

```diff
@@ -1,2 +1,49 @@
 // This component relies on JavaScript from GOV.UK Frontend
 //= require govuk-frontend/components/checkboxes/checkboxes.js
+
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+ (function (Modules) {
+  'use strict'
+   window.GOVUK = window.GOVUK || {};
+   Modules.Checkboxes = function () {
+    this.start = function (scope) {
+        this.initCheckboxChangeListener(scope);
+    };
+
+    this.initCheckboxChangeListener = function(scope) {
+      var _this = this;
+
+      $(scope).find('[data-nested=true] input[type=checkbox]').on('change', function(e) {
+        var checkbox = e.target;
+        var isNested = $(checkbox).closest('.govuk-checkboxes--nested');
+        var hasNested = $('.govuk-checkboxes--nested[data-parent=' + checkbox.id + ']');
+
+        if (hasNested.length) {
+          _this.toggleNestedCheckboxes(hasNested, checkbox);
+        } else if (isNested.length) {
+          _this.toggleParentCheckbox(isNested, checkbox);
+        }
+      });
+    };
+
+    this.toggleNestedCheckboxes = function(scope, checkbox) {
+      if (checkbox.checked) {
+        scope.find('input[type=checkbox]').prop("checked", true);
+      } else if (!checkbox.checked) {
```

Comment:
I think `checkbox.checked` is a boolean so this can just be an else as the only other value it can be is false.